src/ansible_compat/errors.py

```diff
@@ -1,18 +1,49 @@
 """Module to deal with errors."""
+from typing import TYPE_CHECKING, Any, Optional
+
 from ansible_compat.constants import ANSIBLE_MISSING_RC, INVALID_PREREQUISITES_RC
 
+if TYPE_CHECKING:
+    from subprocess import CompletedProcess
+
 
 class AnsibleCompatError(RuntimeError):
     """Generic error originating from ansible_compat library."""
 
     code = 1  # generic error
 
+    def __init__(
+        self, message: Optional[str] = None, proc: Optional[Any] = None
+    ) -> None:
+        """Construct generic library exception."""
+        super().__init__(message)
+        self.proc = proc
+
+
+class AnsibleCommandError(RuntimeError):
```

Comment:
It'd be useful to have a common lib-defined base class for all the exceptions.

Reply:
Additionally, this should also inherit `subprocess.SubprocessError`

Reply:
Actually, I'm not even sure if it should depend on `RuntimeError` because its purpose is to indicate unspecific unexpected things happening at runtime.

Reply:
Clearly not SubprocessError because we may endup executing code with something else (containers) but I am open to suggestions.

Reply:
When you execute things in containers, you still run subprocessed. It's a good abstraction.